packages/parameters/src/types/DynamoDBProvider.ts

```diff
@@ -71,38 +72,120 @@ type DynamoDBProviderOptions =
 /**
  * Options for the DynamoDBProvider get method.
  *
- * @interface DynamoDBGetOptionsInterface
+ * @interface DynamoDBGetOptionsBase
  * @extends {GetOptionsInterface}
  * @property {number} maxAge - Maximum age of the value in the cache, in seconds.
  * @property {boolean} forceFetch - Force fetch the value from the parameter store, ignoring the cache.
  * @property {GetItemCommandInput} [sdkOptions] - Additional options to pass to the AWS SDK v3 client.
  * @property {TransformOptions} transform - Transform to be applied, can be 'json' or 'binary'.
  */
-interface DynamoDBGetOptionsInterface extends GetOptionsInterface {
+interface DynamoDBGetOptionsBase extends GetOptionsInterface {
   sdkOptions?: Omit<
     Partial<GetItemCommandInput>,
     'Key' | 'TableName' | 'ProjectionExpression'
   >;
 }
 
+interface DynamoDBGetOptionsTransformJson extends DynamoDBGetOptionsBase {
+  transform: 'json';
+}
+
+interface DynamoDBGetOptionsTransformBinary extends DynamoDBGetOptionsBase {
+  transform: 'binary';
+}
+
+interface DynamoDBGetOptionsTransformNone extends DynamoDBGetOptionsBase {
+  transform?: never;
+}
+
+type DynamoDBGetOptions =
+  | DynamoDBGetOptionsTransformNone
+  | DynamoDBGetOptionsTransformJson
+  | DynamoDBGetOptionsTransformBinary
+  | undefined;
+
+/**
+ * Generic output type for DynamoDBProvider get method.
+ */
+type DynamoDBGetOutput<
```

Comment:
Quite an artistic way for a type definition, forces one to reread multiple times. 

So I will give a try to read the priority:

1. ExplicitUserProvidedType, if not undefined
2. InferredFromOptionsType, if not undefined
   2.1 JSONValue, if undefined or TransformJson or TransformNone
   2.2 String, if TransformBinary
   2.3 never, else
   
Is that right?

Reply:
That's correct!

2.3 should never happen at runtime, hence the `never`.

As a side note, I agree that these types are on the ugly side. Unfortunately at the moment I don't have a better way of expressing them.